presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkBlockSerde.java

```diff
@@ -245,44 +237,38 @@ public Slice getDataSource()
         }
     }
 
-    @State(Thread)
-    public static class LongDecimalNoNullBenchmarkData
+    public abstract static class TypeBenchmarkData<T>
             extends BenchmarkData
     {
-        @Setup
-        public void setup()
-                throws Exception
+        private final Type type;
+        private final Function<Random, T> valueGenerator;
+        private final BiConsumer<BlockBuilder, T> blockWriter;
+
+        public TypeBenchmarkData(Type type, Function<Random, T> valueGenerator, BiConsumer<BlockBuilder, T> blockWriter)
         {
-            setup(LONG_DECIMAL_TYPE, createValues());
+            this.type = requireNonNull(type, "type is null");
+            this.valueGenerator = requireNonNull(valueGenerator, "valueGenerator is null");
+            this.blockWriter = requireNonNull(blockWriter, "blockWriter is null");
         }
 
-        private Iterator<?> createValues()
+        protected double getNullChance()
```

Comment:
why not just pass null change as `TypeBenchmarkData` param?

Reply:
Param annotation only work with fields (I think) and fields are not available in constructor.